packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -1028,11 +1030,12 @@ contract Deploy is Deployer {
         GameType _gameType,
         Claim _absolutePrestate,
         IBigStepper _faultVm,
-        uint256 _maxGameDepth
+        uint256 _maxGameDepth,
+        bool _allowUpgrade
```

Comment:
hmm, this is potentially an issue. If the game implementation for a game code can be changed, the challenger needs some way to tell which version is actually in use. That used to be the game type but this breaks that assumption. It's not enough to just time upgrading the challenger with deploying the new contracts because there may be type 0 games that already exist using the old implementation.

Makes me realise that the challenger can't currently handle network upgrades because it really needs to support having multiple absolute prestates for the before and after op-program versions.

Gut feel here is that we need to prevent changing the meaning of an existing game type and every new version gets a new game type. Then we probably need to make game type bigger than uint8.

The other potential option is that the challenger determines which prestate to use for a set its given based on the absolute prestate claim but that still means the VM could be upgraded (as opposed to just the op-program prestate) and the challenger wouldn't be able to deal with that.

Reply:
Created https://github.com/ethereum-optimism/client-pod/issues/446 to ensure we don't forget to think this all through properly.

Reply:
Resolving this for now and will stick with upgradable until we work through the actual upgrade cases and decide what the best approach is.